lib/cpp/src/thrift/server/TThreadedServer.h

```diff
@@ -20,120 +20,86 @@
-                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory))
+    : TServer(processorFactory, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(new concurrency::PlatformThreadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  const boost::shared_ptr<ThreadFactory>& threadFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
-  virtual ~TThreadedServer();
+  virtual ~TThreadedServer() {};
 
   virtual void serve();
 
-  void stop() {
+  virtual void stop() {
     stop_ = true;
     serverTransport_->interrupt();
   }
 
 protected:
-  void init();
+  virtual void disposeClient(TConnectedClient *pClient);
 
-  boost::shared_ptr<ThreadFactory> threadFactory_;
+  boost::shared_ptr<concurrency::ThreadFactory> threadFactory_;
```

Comment:
My personal belief is that namespace aliases and using clauses lead to really nasty compiler issues over time which is why I removed them, or in this case it seems that is was simply assumed someone else would do this before including the header.  To be safe, perhaps they should all be fully qualified.  I struggled with whether or not to do this because of the verbosity.